From a88aa85fc1f26082f0050e2c1386e7ab4d5e6acd Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.xyz>
Date: Tue, 9 Aug 2016 13:11:18 +0800
Subject: [PATCH] x86/efi-bgrt: remove the check of the version field

Some broken firmwares have a wrongly filled version field in BGRT table.
(See http://wiki.osdev.org/Broken_UEFI_implementations )

As we know, these firmwares can also provide correct BGRT image, although
the table is wrong.

After removing the check of the version field, the kernel can now extract
the image correctly, and the information is also correct.

Tested on a Thinkpad E531 (68854UC).

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>
---
 arch/x86/platform/efi/efi-bgrt.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/x86/platform/efi/efi-bgrt.c b/arch/x86/platform/efi/efi-bgrt.c
index 6a2f569..f492ea0 100644
--- a/arch/x86/platform/efi/efi-bgrt.c
+++ b/arch/x86/platform/efi/efi-bgrt.c
@@ -47,11 +47,6 @@ void __init efi_bgrt_init(void)
 		       bgrt_tab->header.length, sizeof(*bgrt_tab));
 		return;
 	}
-	if (bgrt_tab->version != 1) {
-		pr_notice("Ignoring BGRT: invalid version %u (expected 1)\n",
-		       bgrt_tab->version);
-		return;
-	}
 	if (bgrt_tab->status & 0xfe) {
 		pr_notice("Ignoring BGRT: reserved status bits are non-zero %u\n",
 		       bgrt_tab->status);
-- 
2.9.2

